### PR TITLE
Implement domain layer

### PR DESCRIPTION
**Pull request by timreimherr**
When I changed the branch name it messed up the pull request, and now the first pull request is closed. So, I opened this request. 

Why did you say to make the method in the UIConverter class a static method? Forgot to change it to that and just remembered now, but I will change it to a static. 

I worked on having the BlogPost Controller use the new Domain Layer class and UIConverter class in creating a new post, thus, in order to get the project to build, I ended up persisting all the changes necessary to the View. But now it builds and I can see the website. I'm not sure if I set up the dependencies correctly...

### CODE REVIEW

File: Blog/GamingGuruBlog.Web/GamingGuruBlog.Web/UIConverter.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+using System.Web;
+using GamingGuruBlog.Domain;
+using GamingGuruBlog.Domain.Interfaces;
+using GamingGuruBlog.Domain.Models;
+using GamingGuruBlog.Web.Models;
+using System.Web.Mvc;
+
+namespace GamingGuruBlog.Web
+{
+    public class UIConverter
```
Comment (wojonet) [inline Blog/GamingGuruBlog.Web/GamingGuruBlog.Web/UIConverter.cs]: All this class should do is convert a viewmodel to a domain model and visa versa.

File: Blog/GamingGuruBlog.Web/GamingGuruBlog.Web/UIConverter.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        private IBlogServices _blogServices;
+
+        public UIConverter(IBlogServices _passedInServices)
+        {
+            _blogServices = _passedInServices;
+        }
+
+
+        public BlogPostVM CreateNewBlogPostVM(string userID)
+        {
```
Comment (wojonet) [inline Blog/GamingGuruBlog.Web/GamingGuruBlog.Web/UIConverter.cs]: This work should be performed in the controller.

**Comment by wojonet** [comment]
@timreimherr UIConverter should be static because there's no advantage to having multiple instances of the method that does the conversion.

Everything else looks fine for now. At some point we should sit down and go through the project.

**Review by wojonet — CHANGES_REQUESTED** [review]

File: Blog/GamingGuruBlog.Web/GamingGuruBlog.Web/UIConverter.cs (hunk 1 above)
Comment (timreimherr) [reply Blog/GamingGuruBlog.Web/GamingGuruBlog.Web/UIConverter.cs]: @wojonet , I made these changes and commited them. 
Okay, I may have over done it but I'm not sure. When a BlogPostVM is turned to a BlogPost a lot items are created in the DB and then associated with that BlogPost. So, this is what is happening in the conversion and at the end it returns a BlogPost with all the associations. Is this correct?

**Comment by timreimherr** [comment]
@wojonet 
Okay, I may have over done it, but I'm not sure. When a BlogPostVM is converted to a BlogPost there are items in the BPVM that need to created in the DB in order for them to be associated with the BlogPost that is returned. So that is what the converter class is doing. 

Is that correct?

**Comment by wojonet** [comment]
Yeah I think you're getting it for the most part.....we'll meet soon and go over it, but  I think you have the basic idea. Nice work 👍

**Pull request merged**